src/OpenTelemetry/Trace/Configuration/OpenTelemetrySdk.cs

```diff
@@ -65,9 +68,40 @@ public static void EnableOpenTelemetry(Action<OpenTelemetryBuilder> configureOpe
                 // or not
                 ShouldListenTo = (activitySource) => openTelemetryBuilder.ActivitySourceNames.Contains(activitySource.Name.ToUpperInvariant()),
 
-                // The following parameters are not used now.
+                // The following parameter is not used now.
                 GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> options) => ActivityDataRequest.AllData,
-                GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> options) => ActivityDataRequest.AllData,
+
+                // This delegate informs ActivitySource about sampling decision.
+                // Following simple behavior is enabled now:
+                // If Sampler returns IsSampled as true, returns ActivityDataRequest.AllDataAndRecorded
+                // This creates Activity and sets its IsAllDataRequested to true.
+                // Library authors can check activity.IsAllDataRequested and avoid
+                // doing any additional telemetry population.
+                // Activity.IsAllDataRequested is the equivalent of Span.IsRecording
+                //
+                // If Sampler returns IsSampled as false, returns ActivityDataRequest.None
+                // This prevents Activity from being created at all.
+                GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> options) =>
+                {
+                    var shouldSample = sampler.ShouldSample(
```

Comment:
@cijothomas per the discussion about SpanId we need to do some code here before passing it to the sampler, but that comes into play when we add the probabilistic sampler so I think it is fine to merge as it is and do a follow up PR.

Reply:
If you don't mind I would like to do that so I play with the aspects we talked regarding `SpanId`.

Reply:
Thanks - appreciate your work on it.